module-4/backend/week_four.md

```diff
@@ -1,11 +1,58 @@
 ## Week Four - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
-TBD  
 
-#### Review  
+# Week Four - Module 4 Recap
 
-TBD
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
+
+Note: When you're done, submit a PR.
+
+    What's the purpose of a project management tool?
+      To keep track of a project: what is done, what is being worked on, what needs to be worked on, who is working on what.
+    What are the stages of a card when using Pivotal Tracker?
+      gathering requirements
+      story breakdown
+      story estimation
+      prioritization
+      coding out the story
+      story acceptance
+
+    What are the four types of cards available in Pivotal Tracker?
+      current stories
+      icebox
+      backlog
+      done
+    What's your favorite project management tool?
+      Waffle
+    Why do we create staging environments?
+      So that we can make sure everything is working properly in fake production before putting it to actual production
+    What are the characteristics of a good README (in your opinion)?
+      Have a setup, a quick start, good examples of how to program is used, what the program does, possible endpoints, notes on things that could be difficult, and all docs are explained as if a beginner is reading them.
+    What's one main improvement you're going to make to your code regarding accessibility issues?
+      make this tab-able
+    What are some basic security concerns to be aware of when building applications?
+      making sure you can't access content you should when you log in
+      password storage
+      unencrypted traffic
+      search bar and url - sql injectable
+      faulty authentication
+    Why is continuous integration helpful/important?
+      so that as you build you continue to make the iteration loop - make, refactor, review, redo - and so as you go your making the code better now waiting til the end.
```

Comment:
continuous integration is helpful because it automatically builds your application and runs your tests. this way, you can tell if anything is breaking before pushing it to production